File: _src/04/Chapter4.VB/Maps/PersonTypeMap.vb
Imports System.Data.Entity.ModelConfiguration
Imports Chapter4.VB.Models

Namespace Maps
    Public Class PersonTypeMap
        Inherits EntityTypeConfiguration(Of PersonType)
        Public Sub New()
            HasMany(Function(pt) pt.Persons) _
                .WithOptional(Function(p) p.PersonType) _
                .HasForeignKey(Function(pt) pt.PersonTypeId) _
                .WillCascadeOnDelete(False)
        End Sub
    End Class
End NameSpace

File: _src/04/Chapter4.CSharp/Maps/PersonTypeMap.cs
using System.Data.Entity.ModelConfiguration;
using Chapter4.CSharp.Models;

namespace Chapter4.CSharp.Maps
{
    public class PersonTypeMap : EntityTypeConfiguration<PersonType>
    {
        public PersonTypeMap()
        {
            HasMany(pt => pt.Persons)
                .WithOptional(p => p.PersonType)
                .HasForeignKey(p => p.PersonTypeId)
                .WillCascadeOnDelete(false);
        }
    }
}